### ISSUE TITLE
Can blitz through a friendly neutral. Should not be possible.

### ISSUE THREAD

**Issue by simon33-2** [triplea-game/triplea#4685]
### Engine version
All, 1.9-13066

### My Operating System
Ubuntu

### Map name
World War II Global 2nd Edition

### Can you describe how to trigger the error? (eg: what sequence of actions will recreate it?)
For example, start new game, click through to German NCM, move tank from Romania to Albania via Bulgaria. The same thing can be achieved with a mech infantry.

### Instead of this error, what should have happened?
This should not be possible. When activating a friendly neutral, units must stop in the territory which is activated.

### Any additional information that may help
Europe 1940 2nd ed rules, p10, refer bolded bit:
"Friendly neutrals"
Friendly neutrals are neutral territories that are
sympathetic to your alliance. For example, on the map
Bulgaria and Iraq are shown as being pro-Axis. These
territories would be friendly neutrals to both Germany
and Italy. A pro-Allies neutral would be considered a
friendly neutral to the United States, United Kingdom,
Soviet Union, and France.
Friendly neutrals may not be attacked, and air units
may not fly over them. They can be moved into (**but not
through**) as a noncombat move by land units of a power
that is at war (see “Noncombat Move,” page 22). This
moves the territory out of its neutral status, however. The
first friendly power to do so places its national control
marker on the former friendly neutral territory, and its
national production level is adjusted upward by the value
of the territory. With the territory’s loss of neutrality in
this way, its standing army is immediately activated.
The units placed belong to the power that now controls
the formerly neutral territory, and may be used freely
beginning on that power’s next turn.

**Comment by WCSumpton** [comment]
Right now when 'canMoveLandUnitsOverOwnedLand' is false, land units are prevented for entering owned land territories, when is should only prevent the movement over/through those territories

**Comment by WCSumpton** [comment]
This could be done by using 'territioyEffect' to increase the movement value to Bulgaria during Germany's and Italy's NCM while removing/changing the 'territoryEffect' after their NCM. A trigger would check the owner of Bulgaria and if it was still the friendly neutral set the effect. This could also be used to restrict AA Guns from moving there during the NCM.

**Comment by simon33-2** [comment]
@WCSumpton I'm not sure I understand. Would changing those attributes for Bulgaria prevent moving through the territory after the territory is activated? If so, that would be wrong.

**Comment by WCSumpton** [comment]
@simon33-2 No prior to the activation of the friendly neutral territory, so that unit that have 2 movement will not be able to move through/blitz the territory.

**Comment by Cernelius** [comment]
@simon33-2 What @WCSumpton is proposing would work, as you would have a trigger removing that effect after take over, but it is substantially a hack, so I don't think you want to "fix" this game that way, also considering that this would be purely a map level way to do it, requiring changing the xml of every single game affected.

And, anyways, this would not fix this issue on a general standpoint, assuming TripleA, as a program, wants not blitzing through a friendly neutral.

**Comment by VictoryFirst2** [comment]
@WCSumpton @Cernelius Would a viable fix be to make sure "canMoveLandUnitsOverOwnedLand" only allows movement *into* the territory but not *through*? Or will that break other maps? Creating separate option attachments "canMoveLandUnitsIntoOwnedLand" and "canMoveLandUnitsThroughOwnedLand" could be an option to keep options open for map-makers, however that would require an extensive number of XMLs to be updated. Modifying the existing option attachment sounds best to me, but before doing that I'd like to know if that will not cause any unintended behavior in other maps that also use this option attachment.

**Comment by Cernelius** [comment]
@simon33-2 The "map name" (Actually, that is the "game name".) in the opening post should be "World War II Global 1940 Second Edition".

---

@VictoryFirst1:
> [@WCSumpton](https://github.com/WCSumpton) [@Cernelius](https://github.com/Cernelius) Would a viable fix be to make sure "canMoveLandUnitsOverOwnedLand" only allows movement _into_ the territory but not _through_?

I assume that you are considering @WCSumpton's advice, namely:
>Right now when 'canMoveLandUnitsOverOwnedLand' is false, land units are prevented for entering owned land territories, when is should only prevent the movement over/through those territories

Not being a developer, so not being able to go and see what "canMoveLandUnitsOverOwnedLand" does and is supposed to do, my only  way of know that is by testing and by reading documentation, and I'm not testing.



The only documentation of which I'm aware is the following.

https://github.com/triplea-maps/the_pact_of_steel/blob/master/map/games/pact_of_steel_2.xml#L1602
>		canMoveLandUnitsOverOwnedLand			values: "true", "false", or "default". default setting is "default", and "default" means that archeType of war and allied can move into, while neutral can not

Is it correct? (If not, please update that comment.)

Assuming it is correct and that it actually refers only to combat movements (because you certainly cannot non-combat move land units into enemy territories as default) (so the phrase "can move into" should have actually been "can move into but only during a Combat Movement phase if at war"), my understanding is that the option should remain what it is, and it just determines whether or not you can enter territories during combat movement in general (which is always true for every game not having neutrals, and, by neutrals, it is excluded the Neutral player, which is never neutral but always at war with everyone) and during non-combat movement if not at war. Am I surmising correctly? Can a developer please check this and write a better comment for this option (in "Pact of Steel 2")?

I would agree that the option does not do what it says, but I'm guessing that the problem here is that the option is badly named (and having a badly descriptive comment in "Pact of Steel 2").

Assuming everything I said is correct, making impossible to blitz territories of an enemy would certainly cause major problems. That is the fundamental function of blitzing!

Beside, also making sure that combat-moving through one (or more) allied territories to enter (or blitz) one (or more) enemy territories is necessary (canopener anyone?).

@WCSumpton Am I misunderstanding something?

**Comment by Cernelius** [comment]
My suggestion is adding a property for determining whether or not you can non-combat "blitz" (This is not actually a blitzing move.) as described in the opening post (as well as the program is wrongly letting you do at the current state). The property should be that you cannot as default. This way, the relevant games would be fixed, and any (I assume custom) game having the current behaviour can be updated as to keep it, plus any future map-maker will be able to choose this behaviour (one way or the other) for his own game (which is always good).

CC: @panther2

**Comment by panther2** [comment]
As @Cernelius  implies, the move in question is incorrectly worded as "blitz" in this issue.

The one and only situation that is to be called "blitz" is a move through an unoccupied hostile territory as the first part of a 2-space move that can end in a friendly or hostile territory - during Combat Move Phase only.

In this issue we talk about is simply a 'standard' move (during Non-Combat Move Phase) of a two-movement-points-unit, that moves into a friendly-neutral territory - and is illegally allowed to continue its movement from there.

**Comment by VictoryFirst2** [comment]
I think the only way to solve this is to remove canMoveUnitsOverOwnedLand and create two new relationship properties: canMoveIntoOwnedLand and canMoveThroughOwnedLand, each for air and for land units.

This is because sometimes, you can't even move into the territory (in the case of true neutrals) and sometimes you can only not move through the territory (in the case of friendly neutrals or flying over unfriendly/true neutrals).

Also, some code needs to be added that prevents the player from moving through a territory when he splits up the movement. For example, when Germany first moves to Bulgaria and then to Albania, the second movement is from a friendly territory to another friendly territory, which is allowed by the game. A check needs to be added to see if the unit moved into a friendly neutral before moving again.

Also, a bunch of XMLs would need to be updated as they are relying on canMoveUnitsOverOwnedLand which should technically say canMoveUnitsIntoOwnedLand.

In short, this is not easy to fix.
